themes/helm/layouts/blog/list.html

```diff
@@ -19,7 +19,16 @@
         <article>
           <header>
             <h1><a href="{{ .Permalink }}">{{ .Title }}</a></h1>
-            <p class="date">{{ .Date.Format "Mon, Jan 2, 2006" }}</p>
+            {{ $date := .Date.Format "Mon, Jan 2, 2006" }}
+            {{ $lastmod := .Lastmod.Format "Mon, Jan 2, 2006" }}
+            <p class="date">
+              {{ if and (ne $lastmod $date) (gt .Lastmod .Date) }}
```

Comment:
A note on why: the first condition checks that the date strings aren’t equal (to the day). The second checks the updated timestamp is greater than when it was posted.